core/src/main/java/com/linecorp/armeria/internal/common/util/TransportTypeProvider.java

```diff
@@ -209,16 +170,19 @@ private static <T> Class<T> findClass(String channelPackageName, String classNam
                                         TransportTypeProvider.class.getClassLoader());
     }
 
-    private static BiFunction<Integer, ThreadFactory, ? extends EventLoopGroup> findEventLoopGroupConstructor(
-            Class<? extends EventLoopGroup> eventLoopGroupType) throws Exception {
-        requireNonNull(eventLoopGroupType, "eventLoopGroupType");
-        final MethodHandle constructor =
-                MethodHandles.lookup().unreflectConstructor(
-                        eventLoopGroupType.getConstructor(int.class, ThreadFactory.class));
+    private static BiFunction<Integer, ThreadFactory, ? extends IoEventLoopGroup> findEventLoopGroupFactory(
+            Class<? extends IoHandler> ioHandlerType) throws Exception {
+        requireNonNull(ioHandlerType, "ioHandlerType");
 
+        // Create a new IoHandlerFactory by invoking the factory methods in the ioHandlerType such as
+        // EpollIoHandler.newFactory().
+        final MethodHandle factoryMethod =
+                MethodHandles.lookup().findStatic(ioHandlerType, "newFactory",
+                                                  MethodType.methodType(IoHandlerFactory.class));
         return (nThreads, threadFactory) -> {
             try {
-                return (EventLoopGroup) constructor.invoke(nThreads, threadFactory);
+                final IoHandlerFactory ioHandlerFactory = (IoHandlerFactory) factoryMethod.invoke();
+                return new MultiThreadIoEventLoopGroup(nThreads, threadFactory, ioHandlerFactory);
```

Comment:
`MultiThreadIoEventLoopGroup` and `IoHandlerFactory` are now used to create an `EventLoopGroup`.
https://netty.io/news/2025/04/03/4-2-0.html#:~:text=MultiThreadIoEventLoopGroup,are%20running%20on)